homeassistant/components/deconz/sensor.py

```diff
@@ -67,171 +70,163 @@
 ATTR_EVENT_ID = "event_id"
 
 
+T = TypeVar(
+    "T",
+    AirQuality,
+    Consumption,
+    Daylight,
+    GenericStatus,
+    Humidity,
+    LightLevel,
+    Power,
+    Pressure,
+    Temperature,
+    Time,
+    PydeconzSensorBase,
+)
+
+
 @dataclass
-class DeconzSensorDescriptionMixin:
+class DeconzSensorDescriptionMixin(Generic[T]):
     """Required values when describing secondary sensor attributes."""
 
+    isinstance_fn: Callable[[T], bool]
```

Comment:
I think if you adjust this to be 
```python
    instance_check: type
```

Then you can simplify the entity descriptions, and use `isinstance` inside the loop instead:
```python
            if not isinstance(description.instance_check, sensor):
                continue
```

You could even make it optional so that `battery` and `internal_temperature` simply skip the `isinstance` check.

Reply:
Making it optional would mean that you can also drop the `common` property, and replace `if description.common` with ìf description.instance_check is None`

Reply:
Very sensible consolidations, thanks!